benchmarks/benchmarks/utils.py

```diff
@@ -1,8 +1,102 @@
+import random
+from collections.abc import Callable
+from dataclasses import dataclass
+
 import pandas as pd
 
 import networkx as nx
 
 
+@dataclass
+class BenchmarkGraph:
+    """A graph used for benchmarking.
+
+    Attributes
+    ----------
+    name : str
+        The name of the graph.
+    graph : Callable[[], nx.Graph]
+        Function to generate the graph
+    """
+
+    name: str
+    graph: Callable[[], nx.Graph]
+
+    @classmethod
+    def from_func(cls, func, *args, **kwargs) -> "BenchmarkGraph":
+        """
+        Create a BenchmarkGraph from a graph-generating function and its arguments.
+
+        Parameters:
+            func (Callable): The graph-generating function.
+            *args: Positional arguments.
+            **kwargs: Keyword arguments.
+
+        Returns:
+            BenchmarkGraph: An instance with the generated name and graph.
+        """
+        return cls(
+            name=_benchmark_name_from_func_call(func, *args, **kwargs),
+            graph=lambda: func(*args, **kwargs),
+        )
+
+
```

Comment:
Can we simplify the benchmark class structure using something like this (along with replacing `BenchmarkGraph.from_func(...)` with `BenchmarkGraph(...)`?
```suggestion
class BenchmarkGraph:
    """A graph used for benchmarking.

    Attributes
    ----------
    name : str
        The name of the graph.
    func: Callable
        Function to generate the graph
    *args and **kwargs:  parameters for the graph creation function.
    """
    def __new__(cls, name, func, *args, **kwargs):
        return cls(
            name = _benchmark_name_from_func_call(func, *args, **kwargs),
            graph = lambda: func(*args, **kwargs)
        )
```

Reply:
Yeah, you are right. I droped the class